src/gamebryo/gamebryosavegame.cpp

```diff
@@ -123,19 +123,52 @@ void GamebryoSaveGame::FileWrapper::setPluginString(StringType type)
   m_PluginString = type;
 }
 
-void readQDataStream(QDataStream& data, void* buff, std::size_t length)
+void GamebryoSaveGame::FileWrapper::readQDataStream(QDataStream& data, void* buff,
+                                                    std::size_t length)
 {
-  int read = data.readRawData(static_cast<char*>(buff), static_cast<int>(length));
-  if (read != length) {
+  int read    = data.readRawData(static_cast<char*>(buff), static_cast<int>(length));
+  bool result = true;
+  if (read != length && m_CompressionType == 1) {
+    bool result = readNextChunk();
+    if (result) {
+      read += data.readRawData(static_cast<char*>(buff) + read,
+                               static_cast<int>(length - read));
+    }
+  }
+  if (read != length || !result) {
     throw std::runtime_error("unexpected end of file");
   }
 }
 
 template <typename T>
-void readQDataStream(QDataStream& data, T& value)
+void GamebryoSaveGame::FileWrapper::readQDataStream(QDataStream& data, T& value)
 {
-  int read = data.readRawData(reinterpret_cast<char*>(&value), sizeof(T));
-  if (read != sizeof(T)) {
+  int read    = data.readRawData(reinterpret_cast<char*>(&value), sizeof(T));
+  bool result = true;
+  if (read != sizeof(T) && m_CompressionType == 1) {
+    result = readNextChunk();
+    if (result) {
+      read +=
+          data.readRawData(reinterpret_cast<char*>(&value) + read, sizeof(T) - read);
+    }
+  }
+  if (read != sizeof(T) || !result) {
+    throw std::runtime_error("unexpected end of file");
+  }
+}
+
+void GamebryoSaveGame::FileWrapper::qDataStreamSkip(QDataStream& data,
```

Comment:
I would call this `skipQDataStream` to be consistent.

Reply:
I mainly felt like it was confusing naming. But I debated this myself.